keps/sig-multicluster/1645-multi-cluster-services-api/README.md

```diff
@@ -664,6 +672,12 @@ endpoints:
 The `ServiceImport.Spec.IP` (VIP) can be used to access this service from within
 this cluster.
 
+
+#### Multi-network scenario
+One of more clusters in a ClusterSet can be running on a discrete network (a non-flat network). An MCS controller can use the `network.k8s.io` `ClusterProperty` to determine if a cluster in a `ClusterSet` is running on a discrete network. In this scenario. Note that the endpoints of the `EndpointSlice` for a cluster on discrete network may only be representative of the pods backing the multi-cluster service and not the real pod addresses.
+##### Known limitation
+In a multi-network scenario where the `EndpointSlice`s do not contain the actual pod addresses, there isn't currently a way (K8s native support) to proportionately distribute the traffic based on the actual number of pods. There is active ongoing work in SIG-Network to add an attribute to represent the number of endpoints for `EndpointSlice`s. This will provide a way for kube-proxy to load balance across `EndpointSlice`s .
```

Comment:
I don't think there's actually a KEP for "weight" in EndpointSlices, is there?  @robscott @JeremyOT @lauralorenz @bowei 

Given the KEP freeze next week, there's still time, but not very much, for v1.24, if someone jumps on this.  Otherwise the earliest possible alpha is 1.25 (summer)

Reply:
@thockin Let me start a draft and share by tomorrow

Reply:
There is not a KEP for weight. The primary concern was that of scalability. EndpointSlices are already updated very frequently. Depending on the implementation, weighting could exponentially increase those updates.

/cc @wojtek-t

Reply:
we have to have it SOMEHOW, right?  I don't see how this can go very far without ...

We might need some more clever design for the API, but we need ... something?

Reply:
It _may_ be a reasonable compromise to have per-slice weight instead of per-endpoint weight. That seems like it could potentially work for this use case while still reasonably limiting writes. For topology aware routing we would likely want different weights depending on where the request was originating from. With that said, per-slice weight is a poor solution when there are a small number of endpoints as it would dramatically increase the number of slices.

Reply:
@robscott Great points, we can discuss more about the tradeoffs over the meeting next week.

Reply:
If we ever want to allow multi-network MCS, we need to revisit this.